Modules/IO/DCMTK/test/itkDCMTKGetDicomTagsTest.cxx

```diff
@@ -0,0 +1,215 @@
+  group = 0x0018;
+  element = 0x0086;
+  itk::int32_t int32Target;
+  if (fileReader.GetElementIS(group, element, int32Target, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << int32Target << std::endl;
+    itk::int32_t expectedVal = 12576; // Expect (12576)"1"
+    ITK_TRY_EXPECT_EQUAL(int32Target, expectedVal);
+  }
+
+  // Acquisition Duration
+  group = 0x0019;
+  element = 0x105a;
+  float floatTarget;
+  if (fileReader.GetElementFL(group, element, floatTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << floatTarget << std::endl;
+    float expectedVal = 1304791694;
+    ITK_TRY_EXPECT_EQUAL(floatTarget, expectedVal);
+  }
+
+  // Locs per 3D slab
+  group = 0x0021;
+  element = 0x1057;
+  if (fileReader.GetElementSL(group, element, int32Target, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << int32Target << std::endl;
+    itk::int32_t expectedVal = 124; // Expect (124)"|"
+    ITK_TRY_EXPECT_EQUAL(int32Target, expectedVal);
+  }
+
+  // Samples per Pixel
+  group = 0x0028;
+  element = 0x0002;
+  if (fileReader.GetElementUS(group, element, usTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << usTarget << std::endl;
+    unsigned short expectedVal = 1; // for scalar; 3 for RGB
+    ITK_TRY_EXPECT_EQUAL(usTarget, expectedVal);
+  }
+
+  // Photometric Interpretation
+  group = 0x0028;
+  element = 0x0004;
+  if (fileReader.GetElementUS(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    unsigned short expectedVal = "????";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Rows
+  group = 0x0028;
+  element = 0x0010;
+  unsigned short usTarget;
+  if (fileReader.GetElementUS(group, element, usTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << usTarget << std::endl;
+    unsigned short expectedVal = 256; // Expect(256)""
+    ITK_TRY_EXPECT_EQUAL(usTarget, expectedVal);
+  }
+
+  // Bits Allocated
+  group = 0x0028;
+  element = 0x0100;
+  if (fileReader.GetElementUS(group, element, usTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << usTarget << std::endl;
+    unsigned short expectedVal = 8;
+    ITK_TRY_EXPECT_EQUAL(usTarget, expectedVal);
+  }
+
+  // Unique Image Identifier
+  group = 0x0043;
+  element = 0x1028;
+  if (fileReader.GetElementOB(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "MR Recon <0>";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Scanner Study Entity UID
+  group = 0x0043;
+  element = 0x1061;
+  if (fileReader.GetElementUI(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    outputFile << strTarget << std::endl;
+    std::string expectedVal = "1.2.840.113619.2.133.1762890640.1886.1055165015.961";
+    ITK_TRY_EXPECT_EQUAL(strTarget, expectedVal);
+  }
+
+  // Image Pixel Type
+  IOPixelEnum pixelType = fileReader.GetImagePixelType();
```

Comment:
GetImagePixelType() is obviously too bad and should be fixed or even better removed from the class.

Reply:
The ideal thing would be to fix it. If `itk::DCMTKImageIO` (maybe through `itk::ImageIOBase`) or the `itk::DCMTKImageIO::ReadImageInformation` are doing a better job in this, then part of the code might be valid for here (haven't looked). If we do not find a way to fix it, then it should be deprecated (and yet be tested so that at least its behavior is consistent) prior to being removed.

Reply:
itk::DCMTKImageIO::ReadImageInformation is in fact OK, not ideal too, i don't like that SamplesPerPixel=4 is assumed to be RGBA, in fact it could be e.g. CMYK. But is really not important, currently there are no DICOM files with SamplesPerPixel=4 (CMYK, ARGB), they are deprecated long long ago. So in 99.999% ReadImageInformation work fine. There are more to test (manually first), e.g. how DCMKT handles YBR images, PALETTE_COLOR. Maybe later.

Reply:
> and yet be tested so that at least its behavior is consistent) prior to being removed

test are always good, but not including known completely broken stuff into test suite, sorry. The test files can not match too, sorry.

Reply:
Thanks for all this insight. Will see when I get the time to push this, as I am still being blocked by #2817 to be able to properly do this.

Reply:
Now blocked by #3820.